### ISSUE TITLE
RapiDoc basic auth broken in micronaut-openapi `4.6.0` through `4.8.1`

### ISSUE THREAD

**Issue by chadlwilson** [micronaut-projects/micronaut-openapi#902]
### Expected Behavior

Basic auth should work fine after a username/password is applied in the RapiDoc Authentication panel, for an API declaring such a security scheme :-)

### Actual Behaviour

Currently you'll get 401s for everything - but the root cause is a RapiDoc bug.

RapiDoc is sending a malformed `Authorization` header which omits the required `Basic ` prefix. It is sending `base64(username:password)` only.

This is caused by https://github.com/rapi-doc/RapiDoc/issues/875 which seemingly was ingested due to embedding into micronaut-openapi an unreleased/untagged version of Rapidoc in https://github.com/micronaut-projects/micronaut-openapi/commit/e0c19d4d54d8ac2a679388b2cac7008aca08c0e6

https://github.com/micronaut-projects/micronaut-openapi/blob/786de0040d629166278f77aec65de7f12d865ef1/openapi/src/main/resources/templates/rapidoc/res/rapidoc-min.js#L1

Raising it here for tracking, although understand it needs fixing upstream, unless a downgrade is done to a released version of RapiDoc (or another revision which doesn't have this issue).

### Version

3.8.1

**Comment by altro3** [comment]
@chadlwilson Hi! But this is bug in rapidoc, not in micronaut-openapi. You can use any version of rapidoc. Just set properties like this (set rapidoc version what you need):
```
options.forkOptions.jvmArgs += "-Dmicronaut.openapi.views.spec=rapidoc.enabled=true,rapidoc.copy-resources=false,rapidoc.js.url=https://unpkg.com/rapidoc@9.3.3/dist/"
```

**Comment by altro3** [comment]
Anyway I reverted rapidoc to version 9.3.3 here https://github.com/micronaut-projects/micronaut-openapi/pull/906

**Comment by chadlwilson** [comment]
Yes, I understand it is a bug in Rapidoc which is why I said

> Raising it here for tracking, although understand it needs fixing upstream, unless a downgrade is done to a released version of RapiDoc (or another revision which doesn't have this issue).

Personally I don't think it's good practice to bundle unreleased versions of software (it's not even clear which commit is included in Micronaut which made it a pain for me to track this down) and in a general sense people use libraries/frameworks to avoid having to worry about every more detailed abstraction underneath, and to make opinionated default choices.

On the other hand, Rapidoc doesn't seem to have a consistent release process or automated test approach right now so possibly creates challenges for micronaut.

**Comment by altro3** [comment]
@chadlwilson I completely agree. It was a stupid mistake from me.

**Comment by chadlwilson** [comment]
OK, cool - I'll chase my PR upstream.

Hopefully we can get a release from them soon, it's been quite a while and there seems to have been quite a few changes on `master`.

**Comment by chadlwilson** [comment]
For what it's worth, the issue is fixed in RapiDoc, and now they have released 9.3.4 (which includes the fix): https://github.com/rapi-doc/RapiDoc/releases/tag/v9.3.4

**Comment by altro3** [comment]
@chadlwilson fixed

**Comment by chadlwilson** [comment]
Merged into 4.8.x
